commands/moderation/filter.js

```diff
@@ -0,0 +1,113 @@
+       darn        dern
+       darned      derned
+       darnit      dernit
+       dang        heck
+       frick       fricked
+       crap        omg
+      
+*/
+
+const Discord = require('discord.js');
+
+module.exports = {
+  name: "filter",
+  description: "filter a message",
+
+  execute(msg) {
+    // Make sure user should be filtered
+    if (
+    !msg.member.roles.cache.some(
+      (role) =>
+        role.name === 'Super User' ||
+        role.name === 'Moderator' ||
+        role.name === 'Admin'
+    )
+  ) {
+      // list of words in users message
+      const words = msg.content.toLowerCase().split(' ');
+
+      // regex expression of strong offensive words
+      const strongWords = /hell|damn|goddamn|goddamned|damned|damnit|hell|fuck|fucked|fucking|fucker|fuckwad|fuckwad|motherfucker|wtf|nigger|nigga|penis|vagina|asshole|sex|sexed|piss|pissed|sexy|sexual|sexuality|bastard|bitch|boobs|semen|sperm|jizz|jizzed|whore|prostitute|fornicate|fornication|adultery|adulter|adulteress|slut|buttplug|clitoris|condom|shit[.,/\\:;\*\?\!\@\#\$]?/;
+
+      // regex expression of minor offensive words
+      const lightWords = /darn|dern|darned|derned|darnit|dernit|dang|heck|frick|fricked|crap|omg[.,/\\:;\*\?\!\@\#\$]?/;
+
+      // numbers of strong and minor offensive words
+      let numStrongWords = 0;
+      let numLightWords = 0;
+
+      for (let i = 0; i < words.length; i++) {
+        if (strongWords.test(words[i])) {
+          numStrongWords++;
+        } else if (lightWords.test(words[i])) {
+          numLightWords++
+        }
+      }
+
+      // A single strong word or multiple light words invokes action
+      if (numStrongWords > 0 || numLightWords > 2) {
+        msg.delete();
+
+        const logs = msg.guild.channels.cache.find(channel => channel.name === 'audit-logs')
+
+        dmTheUser(msg);
+        
+        logs.send(`I warned ${msg.author} for saying something hurtful in ${msg.channel}...`);
+        
+        // prepare to call addnote function
+        const command = `cc!addnote ${msg.author} excessive cussing/offensive language in ${msg.channel}.`;
+
+        const addition = ` User said: ${msg.content}`;
+        
+        // call addnote function based on length of user message
+        if (command.length + addition.length <= 255) {
+          logs.send(command + addition);
+        } else {
+          logs.send(command + ' Users message was to long too copy into database.');
+        }
+      }
+    }
+  },
+};
+
+const dmTheUser = (msg) => {
+  // Create an embed, craft it, and DM the user
+  const Embed = new Discord.MessageEmbed()
+    .setColor('#f1d302')
+    .setTitle(`Reminder to ${msg.author.username}`)
+    .setDescription(`Regarding your message in ${msg.channel}:\n\n"${msg.content}"\n\nPlease refrain from excessive cussing and/or posting offensive content.`)
+    .setTimestamp()
+    .setFooter(`${msg.guild.name}`);
+  msg.author.send(Embed);
+}
```

Comment:
I like that this is separated out into its own function.  I think most of the rest of `execute` could also be refactored into functions to make it easier to glance at and see the gist of what is happening.  For example, there could be a `swearCounter` function and a `addNoteAndLog` function, or whatever names you feel work.

This isn't a must for this PR - it could be done in a separate one if you want.

Reply:
Went ahead and split everything up